software/firmware/europi.py

```diff
@@ -298,16 +298,17 @@ def _bounce_wrapper(self, pin):
                 return
             self.last_rising_ms = time.ticks_ms()
             return self._rising_handler()
-        elif self.value() == LOW:
+        else:
             if time.ticks_diff(time.ticks_ms(), self.last_falling_ms) < self.debounce_delay:
                 return
             self.last_falling_ms = time.ticks_ms()
 
-            # Check if 'other' pin is set and if 'other' pins is high and if this pin has been high for long enough.
+            # Check if 'other' pin was set not long ago and still is set and this pin has been high for long enough.

```

Comment:
```suggestion
            # When this button is released after a short press, check if 'other' button is set and currently pressed not long ago.
```
I reworded the comment to me more clear. The main idea I wanted to keep in tact was "set" means the "other" button was defined (currently the only place `_other` gets set is  `_handler_both()` in bootloader.py: https://github.com/Allen-Synthesis/EuroPi/search?q=_handler_both). Let me know if this change accurately captures your intentions with the new additions.